### DISCUSSION TITLE
Is there a way to use previous values on an update mutation?

### DISCUSSION THREAD

**Discussion by iledan-karl** [prisma/prisma#2531] [Q&A] [answered]
Without querying the previous one that is. Like if I want to just add 1 to the value of a column.

Is raw query the only way? If yes, does raw trigger @updatedAt or subscriptions?

Thank you.

**Comment by ryands17** [comment, marked as answer]
Hey @Endergineer :wave: 
Currently `prisma.raw` is the only way possible to do these type of updates. You can track the spec [here](https://github.com/prisma/specs/issues/356) for the implementation.

Using raw it would be something like this:
```ts
await prisma.raw`update post set "views" = "views" + 1 where id = 1`
```
Regarding this
> Is raw query the only way? If yes, does raw trigger @updatedAt or subscriptions?

The `@updatedAt` will be triggered as it's added at the DB level. As for subscriptions, if they are added at the DB level then it will be triggered, but if it's implemented in your business logic, then it needs to be triggered manually after the update.

**Reply by iledan-karl** [reply]
Thanks for the comprehensive answer. Is prisma's built-in subscription implemented at the DB level?

**Reply by ryands17** [reply]
Prisma 2 doesn't have built-in subscriptions, so currently you would have to manually add subscriptions at the DB level or at the business logic level.
You can currently track the progress [here](https://github.com/prisma/prisma/issues/298).

**Reply by iledan-karl** [reply]
I see, thanks again.

**Reply by ryands17** [reply]
Hey @Endergineer :wave: 
A subscriptions [example](https://github.com/prisma/prisma-examples/tree/master/typescript/subscriptions-pubsub) has been added to the Prisma examples repo :smile:

**Comment by TkinterinShanghai** [comment]
This changed, now you can do this:

```
const updatePosts = await prisma.post.updateMany({
  data: {
    views: {
      increment: 1,
    },
    likes: {
      increment: 1,
    },
  },
})
```
https://www.prisma.io/docs/concepts/components/prisma-client/crud#update

**Reply by M98abas** [reply]
As @TkinterinShanghai replay it's work for me

**Reply by smohammadhn** [reply]
Indeed works, Thanks 🍾

**Comment by ludralph** [comment]
Hi @iledan-karl,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
